src/pybamm/expression_tree/symbol.py

```diff
@@ -356,6 +357,12 @@ def domain(self, domain):
         )
 
     @property
+    @deprecated(
+        deprecated_in="25.1.0",
+        removed_in="26.0.0",
+        current_version=pybamm.__version__,
+        details="Use `symbol.domains` instead.",
+    )
```

Comment:
As we now have the decorator to issue the error, could you drop the error that exists inside the function?